### PR TITLE
Fix cg part priority

### PR DESCRIPTION
**Pull request by bigfoodK**
- Fix cg part variants to be rendered from last
  - first item of part_variants will be render on top

### CODE REVIEW

File: luda-editor/client/src/pages/sequence_edit_page/loaded/components/character_editor/render/part_picker.rs [hunk 1]
```diff
@@ -266,11 +271,20 @@ fn render_thumbnail<'a>(
...
                         .map(|variant| variant.id)
                         .collect::<Vec<_>>();
+                    let cg_variant_id_index_map = HashMap::from_iter(
+                        cg_file
+                            .parts
+                            .iter()
+                            .flat_map(|part| part.variants.iter().map(|variant| variant.id))
+                            .enumerate()
+                            .map(|(index, variant_id)| (variant_id, index)),
+                    );
```
Comment (namse) [inline luda-editor/client/src/pages/sequence_edit_page/loaded/components/character_editor/render/part_picker.rs:281]: Why is this needed?

Comment (bigfoodK) [reply luda-editor/client/src/pages/sequence_edit_page/loaded/components/character_editor/render/part_picker.rs:281]: To sort part variants to follow the order in cg_file.
```rust
fn sort_variations(
    part_variants: &mut Vec<(Uuid, Rect<Percent>)>,
    cg_variant_id_index_map: &HashMap<Uuid, usize>,
) {
    part_variants.sort_by(|(a, _), (b, _)| {
        let a_index = cg_variant_id_index_map.get(a).unwrap_or(&0);
        let b_index = cg_variant_id_index_map.get(b).unwrap_or(&0);
        a_index.cmp(b_index)
    });
}
```

Comment (namse) [reply luda-editor/client/src/pages/sequence_edit_page/loaded/components/character_editor/render/part_picker.rs:281]: I mean, why sort is needed? doesn't server give sorted data?

Comment (bigfoodK) [reply luda-editor/client/src/pages/sequence_edit_page/loaded/components/character_editor/render/part_picker.rs:281]: yes, sever gives sorted data, CgFile.
Sort means "sort `cg.part_variants` in sequence to follow CgFile's order"

Comment (namse) [reply luda-editor/client/src/pages/sequence_edit_page/loaded/components/character_editor/render/part_picker.rs:281]: cg.part_variants are not sorted?

Comment (bigfoodK) [reply luda-editor/client/src/pages/sequence_edit_page/loaded/components/character_editor/render/part_picker.rs:281]: cg.part_variants are not sorted right after it has been modified.
Because part picker modifies cg.part_variants using `push` and `retain` without order.
[relative code](https://github.com/NamseEnt/namseent/pull/548/files/6b8b28c57c3d5874e542ccea2b23feea090215c0#diff-d974301884854b7bdf30fe3ed1b102b4fe50ef3ceb2e65b82239b0e5c10d05dfR290-R329)

Comment (namse) [reply luda-editor/client/src/pages/sequence_edit_page/loaded/components/character_editor/render/part_picker.rs:281]: Hmm. if data sorted at once, no more sort would be unnecessary, isn't it?
I think making guarantee data sorted is very idea.

Comment (bigfoodK) [reply luda-editor/client/src/pages/sequence_edit_page/loaded/components/character_editor/render/part_picker.rs:281]: For example, cg always has all the parts in order and the editor only on and off parts so that order doesn't change?

Comment (bigfoodK) [reply luda-editor/client/src/pages/sequence_edit_page/loaded/components/character_editor/render/part_picker.rs:281]: It happened on modifying Cg.
```rust
// on part variant thumbnail in part picker clicked.
match (selected, selection_type) {
    ..
    (false, rpc::data::PartSelectionType::Single) => {
        ..
        // push part variant to cg. it breaks order
        cg.part_variants.push((
            cg_part_variant.id,
            cg_part_variant.rect,
        ));
    }
}
```

Comment (namse) [reply luda-editor/client/src/pages/sequence_edit_page/loaded/components/character_editor/render/part_picker.rs:281]: HMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMM

Comment (namse) [reply luda-editor/client/src/pages/sequence_edit_page/loaded/components/character_editor/render/part_picker.rs:281]: Let's think about it later.

Comment (namse) [reply luda-editor/client/src/pages/sequence_edit_page/loaded/components/character_editor/render/part_picker.rs:281]: https://github.com/NamseEnt/namseent/issues/556

**Comment by namse** [comment]
@bigfoodK Please resolve conflict

**Review by namse — APPROVED** [review]
